utils/filecache/fsfilecache.go

```diff
@@ -0,0 +1,36 @@
+package filecache
+
+import (
+	"context"
+
+	"github.com/ARM-software/golang-utils/utils/filesystem"
+)
+
+type FsEntryProvider struct {
+	fs       filesystem.FS
+	basePath string
+}
+
+func (p *FsEntryProvider) StoreEntry(ctx context.Context, key string, cacheFilesystem filesystem.FS, cacheDir string) (string, error) {
+	srcPath := filesystem.FilePathJoin(p.fs, p.basePath, key)
+	if _, statErr := filesystem.Stat(srcPath); statErr != nil {
+		return "", statErr
+	}
+
+	destPath := filesystem.FilePathJoin(p.fs, cacheDir, key)
+	cpErr := filesystem.CopyBetweenFS(ctx, p.fs, srcPath, cacheFilesystem, destPath)
```

Comment:
this is not atomic so if a different process/goroutine look for this entry in filesystem. As soon as the file is created it will consider it to be present although the copy may not be complete.

there are two ways to do so, 
- change the copyBetweenFS to be atomic but there may be some weird  corner cases to consider
- make it less prone to race conditions here
tmpKey:=fmt.sprintf("%v-tmp")
destPath := filesystem.FilePathJoin(p.fs, cacheDir, key)
destPathTmp:=filesystem.FilePathJoin(p.fs, cacheDir, tmpKey)
	cpErr := filesystem.CopyBetweenFS(ctx, p.fs, srcPath, cacheFilesystem, destPathTmp)
	if cpErr {return}
	cpErr=cacheFilesystem.Move(ctx, destPathTmp, destPath ) // This would be atomic as in the same filesystem
	if cpErr{ return}